platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/reconnect/BrokenBuilder.java

```diff
@@ -32,9 +32,12 @@ public final class BrokenBuilder implements VirtualDataSourceBuilder<TestKey, Te
     VirtualDataSourceBuilder<TestKey, TestValue> delegate;
 
     int numCallsBeforeThrow = Integer.MAX_VALUE;
-    int numCalls = 0;
     int numTimesToBreak = 0;
-    int numTimesBroken = 0;
+
+    // the following fields are volatile to ensure visibility,
+    // as BreakableDataSource.saveRecords called from multiple threads.
+    volatile int numCalls = 0;
+    volatile int numTimesBroken = 0;
```

Comment:
Why not just make all these 4 fields volatile? There is no need to actually synchronize anything in BreakableDataSource, as its `saveRecords()` may not be called on two threads in parallel. The only issue is variable visibility, and to fix it, it's enough to change all fields to volatiles.

Reply:
>Why not just make all these 4 fields volatile? 
Only `numTimesBroken` and `numCalls` are being changed. But if you see value in marking them all as `volatile`, I'll change that. 

>`saveRecords()`  may not be called on two threads in parallel.
Agreed. I'll add the comment about why the syncrhonized block is not required.